src/compiler/scala/tools/nsc/classpath/ZipAndJarFileLookupFactory.scala

```diff
@@ -181,3 +175,27 @@ object ZipAndJarSourcePathFactory extends ZipAndJarFileLookupFactory {
 
   override protected def createForZipFile(zipFile: AbstractFile): ClassPath = ZipArchiveSourcePath(zipFile.file)
 }
+
+final class FileBasedCache[T] {
+  private case class Stamp(lastModified: FileTime, fileKey: Object)
+  private val cache = collection.mutable.Map.empty[java.nio.file.Path, (Stamp, T)]
+
+  def getOrCreate(path: java.nio.file.Path, create: () => T): T = cache.synchronized {
+    val lastModified = Files.getLastModifiedTime(path)
```

Comment:
it seems you can avoid calling `getLastModifiedTime ` here (which calls `Files.readAttributes`internally) and use `attrs.lastModifiedTime()` instead.

Reply:
this could still be addressed :)

Reply:
The latest commit also uses `attrs.lastModifiedTime()`

Reply:
Sorry, my bad, I mis-read.